frontend/src/libhoclient/host_orchestrator_client.go

```diff
@@ -48,11 +48,26 @@ type ConnectWebRTCOpts struct {
 	LocalICEConfig *wclient.ICEConfig
 }
 
-type BuildAPICredential struct {
+type BuildAPICreds interface {
+	ApplyToHTTPRequest(rb *HTTPRequestBuilder)
+}
+
+type AccessTokenBuildAPICreds struct {
 	AccessToken   string
 	UserProjectID string
 }
 
+func (c *AccessTokenBuildAPICreds) ApplyToHTTPRequest(rb *HTTPRequestBuilder) {
+	const HTTPHeaderBuildAPICreds = "X-Cutf-Host-Orchestrator-BuildAPI-Creds"
+	const HTTPHeaderBuildAPICredsUserProjectID = "X-Cutf-Host-Orchestrator-BuildAPI-Creds-User-Project-ID"
+	if c.AccessToken != "" {
+		rb.AddHeader(HTTPHeaderBuildAPICreds, c.AccessToken)
+		if c.UserProjectID != "" {
```

Comment:
what happens if `AccessToken` is empty and `UserProjectID` is not?

Reply:
the `UserProjectID` value is only relevant if there's an access token.